app/src/main/java/ingsw/group1/findmyphone/log/items/LogItemFormatter.java

```diff
@@ -0,0 +1,192 @@
+package ingsw.group1.findmyphone.log.items;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.drawable.Drawable;
+import android.text.format.DateUtils;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import ingsw.group1.findmyphone.R;
+import ingsw.group1.findmyphone.contacts.SMSContact;
+import ingsw.group1.findmyphone.contacts.SMSContactManager;
+import ingsw.group1.findmyphone.event.EventType;
+import ingsw.group1.findmyphone.event.SMSLogEvent;
+import ingsw.group1.findmyphone.location.GeoPosition;
+
+/**
+ * Factory class to help with formatting of LogItems.
+ * This class hides the formatting logic and provides the LogItems with UI displayable data. If a
+ * name or extra are not available for the event, an empty String is returned.
+ *
+ * @author Riccardo De Zen.
+ */
+public class LogItemFormatter {
+
+    private static final String DEFAULT_NAME = "";
+    private static final String DEFAULT_EXTRA = "";
+    private static final int MY_POSITION_STRING_ID = R.string.my_position_extra;
+    private static final int CONTACT_POSITION_STRING_ID = R.string.contact_position_extra;
+    private static final int MY_RING_STRING_ID = R.string.my_ring_extra;
+    private static final int CONTACT_RING_STRING_ID = R.string.contact_ring_extra;
+    private static final int DATE_FORMAT = SimpleDateFormat.MEDIUM;
+    private static final int TIME_FORMAT = SimpleDateFormat.SHORT;
+
+    private SMSContactManager contacts;
+    private Resources resources;
+    private Map<EventType, Drawable> cachedDrawables;
+
+    public LogItemFormatter(Context context) {
+        contacts = SMSContactManager.getInstance(context);
+        resources = context.getResources();
+        cachedDrawables = EventType.getCachedDrawables(context);
+    }
+
+    /**
+     * Method to format an SMSLogEvent. Name and Address are kept the same. Time is formatted as
+     * a Date. Extra is formatted as specified in {@link LogItemFormatter#formatExtra(SMSLogEvent)}.
+     *
+     * @param eventToFormat The event to format. Unknown type events can not be parsed through
+     *                      this method.
+     * @return The formatted {@link LogItem}. Returns {@code null} if the event is of an unknown
+     * type or if the drawable icon could not be loaded.
+     */
+    @Nullable
+    public LogItem formatItem(SMSLogEvent eventToFormat) {
+        if (eventToFormat.getType() == EventType.UNKNOWN)
+            return null;
+        Drawable appropriateDrawable = cachedDrawables.get(eventToFormat.getType());
+        if (appropriateDrawable == null)
+            return null;
```

Comment:
Here it's really necessary to check twice if it's equal to null?

Reply:
The compiler gives a warning if I don't, because the Map itself is not null but if I were to use a Key (event type) that had no value I'd get a null result, so I just preferred to keep the check

Reply:
I used to do it but then I got rid of it since the Professor said that too many checks for the same things are redundant and that preconditions should be used instead, so I simply specified it in the param comment, if you really think it's necessary I can still change it